src/components/can_cooperation/src/commands/button_press_request.cc

```diff
@@ -0,0 +1,129 @@
+ LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "can_cooperation/commands/button_press_request.h"
+#include "can_cooperation/message_helper.h"
+#include "can_cooperation/can_module_constants.h"
+#include "functional_module/function_ids.h"
+#include "json/json.h"
+
+namespace can_cooperation {
+
+namespace commands {
+
+using namespace json_keys;
+using namespace message_params;
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "ButtonPressRequest")
+
+ButtonPressRequest::ButtonPressRequest(
+    const application_manager::MessagePtr& message,
+    CANModuleInterface& can_module)
+    : BaseCommandRequest(message, can_module) {}
+
+ButtonPressRequest::~ButtonPressRequest() {}
+
+void ButtonPressRequest::Execute() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const Json::Value request_params =
+      MessageHelper::StringToValue(message_->json_message());
+
+  const std::string button_name = request_params[kButtonName].asString();
+  const std::string module_type = request_params[kModuleType].asString();
+  bool module_type_and_button_name = true;
+
+  if (enums_value::kRadio == module_type) {
+    module_type_and_button_name = (button_name == enums_value::kVolumeUp) ||
+                                  (button_name == enums_value::kVolumeDown) ||
+                                  (button_name == enums_value::kEject) ||
+                                  (button_name == enums_value::kSource) ||
+                                  (button_name == enums_value::kShuffle) ||
+                                  (button_name == enums_value::kRepeat);
+  }
+
+  if (enums_value::kClimate == module_type) {
+    module_type_and_button_name = (button_name == enums_value::kACMax) ||
+                                  (button_name == enums_value::kAC) ||
+                                  (button_name == enums_value::kRecirculate) ||
+                                  (button_name == enums_value::kFanUp) ||
+                                  (button_name == enums_value::kFanDown) ||
+                                  (button_name == enums_value::kTempUp) ||
+                                  (button_name == enums_value::kTempDown) ||
+                                  (button_name == enums_value::kDefrostMax) ||
+                                  (button_name == enums_value::kDefrost) ||
+                                  (button_name == enums_value::kDefrostRear) ||
+                                  (button_name == enums_value::kUpperVent) ||
+                                  (button_name == enums_value::kLowerVent);
```

Comment:
@LitvinenkoIra propose to move this comparison snake to a separate class function and then just use
`const bool module_type_and_button_name = SomeFunctionToCheckButtonName();`